docs/proposals/PROPOSAL_H.md

```diff
@@ -0,0 +1,281 @@
+
+      The highest score of the compatibility set means this compatibility set is the most optimal one.
+
+      If this field is not present, the score of this compatibility set is 0.
+
+      If compatibility sets have the same score, that means any of these compatibility sets pass the compatibility check is successful in matching the compatibility requirements.
+
+      The compatibility-score is used for selecting optimal image or optimal node in descending order for those compatibility sets pass the compatibility check.
+
+      The value type of this field is INT.
+
+    - **`annotation field`**
+
+      This OPTIONAL property is a field to describe human readable information regarding to the compatibility set.
+
+      The value type of this field is STRING.
+
+## Artifact example of use
+
+### Example - image is compatible with different hardware combinations
+
+```json
+{
+  "compatibilities": [
+    {
+      "cpu-model.vendor_id": "GenuineIntel",
+      "cpu-cpuid.AVX512FP16": "true",
+      "kernel-selinux.enabled": "false",
+      "kernel-config.PREEMPT": "true",
+      "kernel-version.full": "[4.19, 5.16)",
+      "system.glibc": "[2.31, 2.37]",
+      "pci-15B3-020D.present": "true",
+    },
+    {
+      "cpu-model.vendor_id": "AuthenticAMD",
+      "cpu-cpuid.FPHP": "true",
+      "kernel-selinux.enabled": "false",
+      "kernel-config.PREEMPT": "true",
+      "kernel-version.full": "[4.19, 5.16)",
+      "system.glibc": "[2.31, 2.37]",
+      "pci-1002-67ff.present": "true",
+      "annotation": "only work with AMD CPU and AMD GPU"
+    }
+  ]
+}
+```
+
+### Example - image is compatible with optimal hardware
+
+```json
+{
+  "compatibilities": [
+    {
+      "cpu-model.vendor_id": "GenuineIntel",
+      "cpu-cpuid.AMXFP16": "true",
+      "kernel-selinux.enabled": "false",
+      "kernel-config.PREEMPT": "true",
+      "kernel-version.full": "[4.19, 5.16)",
+      "system.glibc": "[2.31, 2.37]",
+      "system.openmpi": "[4.1.6, 4.1.6]",
+      "pci-15B3-019D.present": "true",
+      "compatibility-score": 50,
+    },
+    {
+      "cpu-model.vendor_id": "GenuineIntel",
+      "cpu-cpuid.AVX512FP16": "true",
+      "kernel-selinux.enabled": "false",
+      "kernel-config.PREEMPT": "true",
+      "kernel-version.full": "[4.19, 5.16)",
+      "system.glibc": "[2.31, 2.37]",
+      "system.openmpi": "[4.1.6, 4.1.6]",
+      "pci-15B3-020D.present": "true",
+      "compatibility-score": 100,
+    },
+    {
+      "cpu-model.vendor_id": "AuthenticAMD",
+      "cpu-cpuid.FPHP": "true",
+      "kernel-selinux.enabled": "false",
+      "kernel-config.PREEMPT": "true",
+      "kernel-version.full": "[4.19, 5.16)",
+      "system.glibc": "[2.31, 2.37]",
+      "system.openmpi": "[4.1.6, 4.1.6]",
+      "pci-1002-67ff.present": "true",
+      "compatibility-score": 80,
+      "annotation": "only work with AMD CPU and AMD GPU"
+    }
+  ]
+}
+```
+
+## Alternative place to hold compatibility set
```

Comment:
@sudo-bmitch the idea here is quite similar as proposal E to reuse platform.features. It's not perfect, but may worth of discussing.

Reply:
We may want to use a different/new field name since there's an implied historical usage of `platform.features`. What happens if both the platform representation and an artifact associated by the referrers API exists? Do runtimes need to query for referrers if the platform field is seen?

Reply:
+1, any existing field have its own original purpose, and once it was published, it may already be  used (for example, os.features used by Unikraft unikernel images) somewhere, even if it's reserved. 

Extending image spec with new field platform.compatibilities may be better idea. the field is a list of map. will update in next commit.

Then it'll be easier to  address these questions:

> What happens if both the platform representation and an artifact associated by the referrers API exists? Do runtimes need to query for referrers if the platform field is seen?

if platform representation is detected, referrs API query will not be sent

Reply:
> We may want to use a different/new field name since there's an implied historical usage of `platform.features`. What happens if both the platform representation and an artifact associated by the referrers API exists? Do runtimes need to query for referrers if the platform field is seen?

The section "Compatibilities Embeded Inside Platform Object" is to extend the platform object,  and talk about migration and deprecation path, which was added in the latest commit. It's inspired by the comment and proposal E.